### PR TITLE
feat(STONEINTG-382): adapt logging to ADR-6 format

### PR DESCRIPTION
**Pull request by MartinBasti**
Updates log format to fulfil [ADR-6](https://issues.redhat.com//browse/ADR-6) format requirements

* JSON format
* Timestamp in expected format
* added caller

https://redhat-appstudio.github.io/book/ADR/0006-log-conventions.html

### CODE REVIEW

**Comment by MartinBasti** [comment]
Example:
```
{"level":"info","ts":"2023-04-20T16:11:39Z","logger":"controllers.integrationTestScenario","caller":"scenario/scenario_adapter.go:93","msg":"IntegrationTestScenario has no environment defined","IntegrationTestScenario":"default/example-pass","Scenario.Name:":"example-pass","Scenario.Namespace:":"default"}
{"level":"info","ts":"2023-04-20T16:11:39Z","logger":"controllers.snapshot","caller":"snapshot/snapshot_adapter.go:161","msg":"The Snapshot has finished testing.","Integration":"default/application-sample-g9glj"}


```

File: controllers/snapshot/snapshot_adapter.go [hunk 1]
```diff
@@ -109,6 +108,11 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (reconciler.Operation
 					a.logger.Error(err, "Failed to create pipelineRun for snapshot and scenario")
 					return reconciler.RequeueOnErrorOrStop(err)
 				}
+				a.logger.Info("IntegrationTestscenario pipeline has been created",
```
Comment (dirgim) [inline controllers/snapshot/snapshot_adapter.go:111]: Should we also add a `a.logger.LogAuditEvent` call with `h.LogActionAdd` action here?

Comment (MartinBasti) [reply controllers/snapshot/snapshot_adapter.go:111]: yes, but I cannot find reference to created pipeline, it may need more code changes

Comment (MartinBasti) [reply controllers/snapshot/snapshot_adapter.go:111]: normally OCP returns object in POST response, but with this k8s runtime client, I don't know how to get name of object when name is generated dynamically

Comment (MartinBasti) [reply controllers/snapshot/snapshot_adapter.go:111]: I was blind :| I'll do it in separate commit

Comment (MartinBasti) [reply controllers/snapshot/snapshot_adapter.go]: added

Comment (dirgim) [reply controllers/snapshot/snapshot_adapter.go]: Brilliant, looks good

**Comment by MartinBasti** [comment]
I pulled main and have to updated `snapshotenvironmentbinding` related files and pipeline_adapter

**Comment by MartinBasti** [comment]
new code covered, ready for review again

**Review by dirgim — APPROVED** [review]
LGTM

File: controllers/binding/snapshotenvironmentbinding_adapter.go [hunk 2]
```diff
@@ -15,15 +15,16 @@ package binding
...
+
 	"github.com/redhat-appstudio/operator-goodies/reconciler"
 	ctrl "sigs.k8s.io/controller-runtime"
 
-	"github.com/go-logr/logr"
 	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
 	"github.com/redhat-appstudio/integration-service/api/v1alpha1"
 	"github.com/redhat-appstudio/integration-service/gitops"
-	"github.com/redhat-appstudio/integration-service/helpers"
+	h "github.com/redhat-appstudio/integration-service/helpers"
```
Comment (hongweiliu17) [inline controllers/binding/snapshotenvironmentbinding_adapter.go:25]: I found in some files the import is change `h "github.com/redhat-appstudio/integration-service/helpers"`,  is there any special reason? Thanks.

**Review by hongweiliu17 — COMMENTED** [review]
I have two questions only. Others looks good to me. Thanks for your work. Learned it!

File: controllers/binding/snapshotenvironmentbinding_adapter.go (hunk 2 above)
Comment (MartinBasti) [reply controllers/binding/snapshotenvironmentbinding_adapter.go:25]: just to make it shorter in the code

File: controllers/pipeline/pipeline_adapter.go [hunk 3]
```diff
@@ -174,19 +175,23 @@ func (a *Adapter) EnsureSnapshotPassedAllTests() (reconciler.OperationResult, er
...
+	if a.component != nil && h.HasLabelWithValue(existingSnapshot, gitops.SnapshotTypeLabel, gitops.SnapshotComponentType) && !gitops.IsSnapshotCreatedByPACPullRequestEvent(existingSnapshot) {
 		compositeSnapshot, err := a.createCompositeSnapshotsIfConflictExists(a.application, a.component, existingSnapshot)
 		if err != nil {
 			a.logger.Error(err, "Failed to determine if a composite snapshot needs to be created because of a conflict",
 				"Snapshot.Name", existingSnapshot.Name)
 			return reconciler.RequeueWithError(err)
 		}
+
 		if compositeSnapshot != nil {
+			a.logger.LogAuditEvent("CompositeSnapshot created", compositeSnapshot, h.LogActionAdd)
```
[the remark this answers is not in the document]
Comment (MartinBasti) [reply controllers/pipeline/pipeline_adapter.go]: Thanks! I moved audit log about creation into function `createCompositeSnapshotsIfConflictExists`

File: controllers/binding/snapshotenvironmentbinding_adapter.go (hunk 2 above)
Comment (hongweiliu17) [reply controllers/binding/snapshotenvironmentbinding_adapter.go:25]: I guess it is the reason. Just notice in some file it still use the old style, such as https://github.com/redhat-appstudio/integration-service/pull/160/files#diff-8147062616b315a0724a1e97be5d43ac179afd264b847c27fb31d21842f308f3R34, so I wonder. But it doesn't matter. Thanks for the explanation.

File: controllers/pipeline/pipeline_adapter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (hongweiliu17) [reply controllers/pipeline/pipeline_adapter.go]: Thank you for the update!

**Review by hongweiliu17 — APPROVED** [review]

**Pull request merged**